static/js/publisher/form.js

```diff
@@ -39,11 +40,35 @@ function initSnapIconEdit(iconElId, iconInputId, state) {
     });
 
     updateState(state, { images });
+    removeIconEl.classList.remove("u-hide");
   });
 
-  snapIconEl.addEventListener("click", function() {
+  snapIconEl.addEventListener("click", function(e) {
+    e.preventDefault();
     snapIconInput.click();
   });
+
+  removeIconEl.addEventListener("click", function(e) {
+    e.preventDefault();
+    snapIconEl.src =
+      "https://assets.ubuntu.com/v1/be6eb412-snapcraft-missing-icon.svg";
+
+    const images = state.images.filter(image => image.type !== "icon");
+
+    updateState(state, { images });
+
+    snapIconInput.value = "";
+    snapIconInput.closest("form").dispatchEvent(
```

Comment:
Why do we need to dispatch event manually here? Especially "change" event on a form object?

Reply:
Good point, I should just `updateState` :+1:

Reply:
... It's already `updateState`ing so yeah can remove :+1::+1:

Reply:
Not sure what you want to trigger here. There is some function (`checkForm`?) to update the Revert/Submit buttons state based on form state. But I'm not sure if `updateState` doesn't to it internally as well...

Reply:
I'm not sure either now - I think because I was resetting the value I needed to cause the form to update